proofs/sectorbuilder/transforms.go

```diff
@@ -0,0 +1,94 @@
+package sectorbuilder
+
+import (
+	"unsafe"
+
+	"github.com/filecoin-project/go-filecoin/types"
+	"github.com/ipfs/go-cid"
+)
+
+// #cgo LDFLAGS: -L${SRCDIR}/../lib -lfilecoin_proofs
+// #cgo pkg-config: ${SRCDIR}/../lib/pkgconfig/libfilecoin_proofs.pc
+// #include "../include/libfilecoin_proofs.h"
+import "C"
+
+func goBytes(src *C.uint8_t, size C.size_t) []byte {
+	return C.GoBytes(unsafe.Pointer(src), C.int(size))
+}
+
+func goPoStProofs(partitions C.uint8_t, src *C.uint8_t, size C.size_t) ([]types.PoStProof, error) {
```

Comment:
> I start to feel uncomfortable having them private

Unfortunately, I can't export these things (and share them across packages, for instance). This is due to a limitation in CGO:

> Cgo translates C types into equivalent unexported Go types. Because the translations are unexported, a Go package should not expose C types in its exported API: a C type used in one Go package is different from the same C type used in another.

They aren't documented because they aren't exported. They aren't tested because they aren't part of the public (exported) API.

That said, I am to write some documentation for these unexported functions.